lib/src/builder.dart

```diff
@@ -62,19 +62,41 @@ import 'package:source_gen/source_gen.dart';
 /// will produce a "'.mocks.dart' file with such mocks. In this example,
 /// 'foo.mocks.dart' will be created.
 class MockBuilder implements Builder {
+  @override
+  final Map<String, List<String>> buildExtensions = {
+    '.dart': ['.mocks.dart']
+  };
+
+  MockBuilder({Map<String, List<String>>? buildExtensions}) {
+    this.buildExtensions.addAll(buildExtensions ?? {});
+  }
+
   @override
   Future<void> build(BuildStep buildStep) async {
     if (!await buildStep.resolver.isLibrary(buildStep.inputId)) return;
     final entryLib = await buildStep.inputLibrary;
     final sourceLibIsNonNullable = entryLib.isNonNullableByDefault;
-    final mockLibraryAsset = buildStep.inputId.changeExtension('.mocks.dart');
+
+    // While it can be acceptable that we get more than 2 allowedOutputs,
+    // because it's the general one and the user defined one. Having
```

Comment:
I don't understand this - user defined configuration for the `.dart` input will override the default. I would expect that the allowed outputs should always have a single value.

Reply:
Oh I see, the examples are using a `^test` in the build extensions to substitute for a `generate_for`. cc @jakemac53 - did we predict this would happen when we enhanced the build extension syntax?


Maybe it would be better to _only_ do a full override and not do a merge.

```dart
  @override
  final Map<String, List<String>> buildExtensions;

  MockBuilder({Map<String, List<String>>? buildExtensions})
      : buildExtensions = buildExtensions ??
            const {
              '.dart': ['.mocks.dart'],
            };
```

Reply:
Having the commit history would be helpful now ahah, but that was my first implementation, but there's a reason I changed it into a merge. I've discovered this when making the example for this config, `example/build_extensions`, when doing the full override and only having an input pattern that only **partially** covers all `.dart` files, the unmatched `.dart` files wouldn't be processed, even though there's a `generate_for` for those files. So my fix was to merge the user rules with the generic one  ( `.dart`=>`.mocks.dart`)

Reply:
> when doing the full override and only having an input pattern that only **partially** covers all `.dart` files, the unmatched `.dart` files wouldn't be processed, even though there's a `generate_for` for those files.

I think not processing them is preferable to silently falling back on potentially unexpected behavior. I think it is reasonable to expect the overridden config to cover all inputs.

Reply:
> I think not processing them is preferable to silently falling back on potentially unexpected behavior. I think it is reasonable to expect the overridden config to cover all inputs.

I can understand your perspective, in this case it's best to be consistent and not introduce any unexpected behavior, even though I think my approach is more ergonomic for developers. I'll try to change it then.